### PR TITLE
Fix training can specify any label

### LINKED ISSUE #1499: Training will still start and successfully run when specified label not in the table
Training SQL:

```sql
SELECT * FROM housing.train
TO TRAIN DNNRegressor
WITH
  model.hidden_units=[128,32],
  train.epoch = 20,
  train.batch_size=4,
  validation.select="select * from housing.test"
LABEL class
INTO sqlflow_models.my_reg_model;
```

Yet the `LABEL` class is not in `housing.train` table, it should be `target`, but the above SQL can still run.

### PR DESCRIPTION
**Pull request by typhoonzero**
Fix https://github.com/sql-machine-learning/sqlflow/issues/1499

### CODE REVIEW

File: pkg/sql/ir_generator.go [hunk 1]
```diff
@@ -127,6 +137,31 @@ func generateTrainStmtByModel(slct *parser.SQLFlowSelectStmt, connStr, cwd, mode
...
+		return e
+	}
+
+	for _, fc := range trainStmt.Features {
+		for _, field := range fc {
+			for _, fm := range field.GetFieldDesc() {
+				name := fm.Name
+				_, ok := trainFields.get(name)
+				if !ok {
+					return fmt.Errorf("Feature field not exist in database: %s", name)
```
Comment (shendiaomo) [inline pkg/sql/ir_generator.go]: 1. Feature field **doesn't** exist
2. Error strings should not be capitalized https://github.com/golang/go/wiki/CodeReviewComments#error-strings

File: pkg/sql/ir_generator.go [hunk 2]
```diff
@@ -127,6 +137,31 @@ func generateTrainStmtByModel(slct *parser.SQLFlowSelectStmt, connStr, cwd, mode
...
+				}
+			}
+		}
+	}
+	labelFieldName := trainStmt.Label.GetFieldDesc()[0].Name
+	_, ok := trainFields.get(labelFieldName)
+	if !ok {
+		return fmt.Errorf("Label field not exist in database: %s", labelFieldName)
```
Comment (shendiaomo) [inline pkg/sql/ir_generator.go]: ditto

File: pkg/sql/executor_ir_test.go [hunk 3]
```diff
@@ -128,7 +128,7 @@ WITH
 LABEL class
 INTO sqlflow_models.my_xgboost_model_by_program;
 
-SELECT sepal_length as sl, sepal_width as sw FROM iris.test
+SELECT sepal_length as sl, sepal_width as sw, class FROM iris.test
```
Comment (tonyyang-svail) [inline pkg/sql/executor_ir_test.go]: Why updating this line?

File: pkg/sql/ir_generator.go [hunk 4]
```diff
@@ -124,7 +136,38 @@ func generateTrainStmtByModel(slct *parser.SQLFlowSelectStmt, connStr, cwd, mode
 	if err != nil {
 		return nil, err
 	}
-	return generateTrainStmtWithInferredColumns(slctWithTrain, connStr)
+	return generateTrainStmtWithInferredColumns(slctWithTrain, connStr, false)
```
Comment (tonyyang-svail) [inline pkg/sql/ir_generator.go:169]: Why we are not verifying the label field of the trainSQL using prediction?

File: pkg/sql/executor_ir_test.go (hunk 3 above)
Comment (typhoonzero) [reply pkg/sql/executor_ir_test.go]: Removing, it's for debug.

File: pkg/sql/ir_generator.go (hunk 4 above)
Comment (typhoonzero) [reply pkg/sql/ir_generator.go:169]: Predict SQL may not select the label column in the SELECT statement, we need only verify the feature columns only in that case.

**Review by shendiaomo — APPROVED** [review]

**Pull request merged**